Library/Homebrew/cask/installer.rb

```diff
@@ -461,20 +457,28 @@ def uninstall_artifacts(clear: false)
         if artifact.respond_to?(:uninstall_phase)
           odebug "Uninstalling artifact of class #{artifact.class}"
           artifact.uninstall_phase(
-            command: @command, verbose: verbose?, skip: clear, force: force?, upgrade: upgrade?,
+            command:   @command,
+            verbose:   verbose?,
+            skip:      clear,
+            force:     force?,
+            successor: successor,
           )
         end
 
         next unless artifact.respond_to?(:post_uninstall_phase)
 
         odebug "Post-uninstalling artifact of class #{artifact.class}"
         artifact.post_uninstall_phase(
-          command: @command, verbose: verbose?, skip: clear, force: force?, upgrade: upgrade?,
+          command:   @command,
+          verbose:   verbose?,
+          skip:      clear,
+          force:     force?,
+          successor: successor,
         )
       end
     end
 
-    def zap
+    def zap(successor: nil)
```

Comment:
```suggestion
    def zap
```

`zap` will never have a successor.

Reply:
What about `brew reinstall --zap`? When I run that on an already-installed cask, there is a successor.

Reply:
Looking at this again, I realized I forgot to pass `successor` to the `uninstall_artifacts` function. I've fixed that.

Reply:
I didn't realize `--zap` was supported for `reinstall`. I think it still makes sense that `zap` removes an application completely before reinstalling, as that's its intended purpose.